projects/pluto/src/main/java/nl/fairspace/pluto/web/MetadataSourcesResource.java

```diff
@@ -0,0 +1,40 @@
+package nl.fairspace.pluto.web;
+
+import lombok.extern.slf4j.Slf4j;
+import nl.fairspace.pluto.config.dto.PlutoConfig;
+import nl.fairspace.pluto.web.dto.MetadataSourceInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static nl.fairspace.pluto.config.Urls.METADATA_SOURCES_PATH;
+
+@RestController
+@Slf4j
+public class MetadataSourcesResource {
+
+    @Autowired
+    PlutoConfig plutoConfig;
```

Comment:
Can be private I think. Also, no need to put annotation over it, just define a constructor with @AllArgsContructor - spring automatically autowires

Reply:
Are you sure? I haven't tried that yet... Also we do it his way in all other places, so it would be good to change them all then for consistency

Reply:
Also, this is not a standard class, but RestController, there is no standard constructor for it

Reply:
I am sure :) this is a relatively old spring's feature. Also, it is recommended to put the autowired annotation over either constructor or setter as, otherwise, for (at least) private fields it will be using reflection, what slows down the init stage.

Regarding 'standard constructor': from Spring's perspective it is just a regular class, Spring uses constructor you provide for bean creation. Then populate fields (if not yet within constructor with parameters). In this case it will be using no args constructor and then (not sure) set it directly (as it is of default access modifier) or within reflection

Reply:
So, it should work if we put it like this:

```
@RequiredArgsConstructor
public class MetadataSourcesResource {

    private PlutoConfig plutoConfig;
    ....
}
```

Reply:
but you're right @tgreenwood, constructor injection is indeed a recommended way...
I'll just put the constructor there instead of the annotation:
```
    private final PlutoConfig plutoConfig;
  
    public MetadataSourcesResource(PlutoConfig plutoConfig) {
        this.plutoConfig = plutoConfig;
    }
```

Reply:
Sorry, I missed `final` in my piece of code, in this case it would work you with `@RequiredArgsContructor`. And also all fields are supposed to be final too, then it is not a problem

Reply:
but feel free to leave it the way you have put it, not to stuck on this not critical comment